### PR TITLE
Fix S3Client.copy return value consistency

### PR DESCRIPTION
**Pull request by dlstadther**
## Description

Adds support for num/size copy response for single-file copies.

## Motivation and Context

S3Client.copy doesn't return a tuple if not copying a directory. This patch adds support for num/size tuple copy response for non-directories.

## Have you tested this? If so, how?
 

Adds assertions for minimum values to existing copy tests.
All s3_test.py tests pass locally.

### CODE REVIEW

**Comment by dlstadther** [comment]
Just tagging a couple other recent `luigi/contrib/s3.py` contributors to quicken the review process. Your review would be appreciated!

@gardenunez @ouanixi

File: test/contrib/s3_test.py [hunk 1]
```diff
@@ -527,3 +511,11 @@ def _run_multipart_test(self, part_size, file_size, **kwargs):
         key_size = s3_client.get_key(s3_path).size
         self.assertEqual(file_size, key_size)
         tmp_file.close()
+
+    def _run_copy_response_test(self, response):
```
Comment (gardenunez) [inline test/contrib/s3_test.py]: I'd find the way to check that the size is strictly equal to the sum instead of using GreaterEqual. Same thing to the number of items.

File: test/contrib/s3_test.py [hunk 2]
```diff
@@ -454,15 +453,17 @@ def test_copy_dir(self):
             self.assertTrue(s3_client.exists(file_path))
 
         s3_dest = 's3://mybucket/copydir_new/'
-        s3_client.copy(s3_dir, s3_dest, threads=10, part_size=copy_part_size)
+        response = s3_client.copy(s3_dir, s3_dest, threads=10, part_size=copy_part_size)
```
Comment (gardenunez) [inline test/contrib/s3_test.py]: I'd have a test for empty directory and empty file cases.

Comment (dlstadther) [reply test/contrib/s3_test.py]: There already exists a `test_copy_empty_file`, but I'll add a test for empty directory

**Comment by dlstadther** [comment]
@gardenunez Thanks for the review and suggestions!

> Another suggestion would be to extract copy-dir and copy-file logic inside copy to a separate function. Maybe even to have copy_dir and copy_file functions instead of copy, this one I still not sure.

I have no opposition to this refactor, but it is outside of the scope of this PR. Feel free to contribute it separately though :)

File: test/contrib/s3_test.py [hunk 3]
```diff
@@ -425,6 +424,26 @@ def test_copy_empty_file(self):
         """
         self._run_copy_test(self.test_put_multipart_empty_file)
 
+    @mock_s3
+    @skipOnTravis('https://travis-ci.org/spotify/luigi/jobs/145895385')
```
Comment (gardenunez) [inline test/contrib/s3_test.py]: can you elaborate?

File: test/contrib/s3_test.py [hunk 4]
```diff
@@ -527,3 +531,13 @@ def _run_multipart_test(self, part_size, file_size, **kwargs):
         key_size = s3_client.get_key(s3_path).size
         self.assertEqual(file_size, key_size)
         tmp_file.close()
+
+        return 1, key_size
+
+    def _run_copy_response_test(self, response, expected_num=None, expected_size=None):
```
Comment (gardenunez) [inline test/contrib/s3_test.py]: :+1:

File: test/contrib/s3_test.py (hunk 3 above)
Comment (dlstadther) [reply test/contrib/s3_test.py]: Sorry. Copy/paste. Will remove to try to green build without it.

File: test/contrib/s3_test.py [hunk 5]
```diff
@@ -235,7 +235,7 @@ def test_put_multipart_less_than_split_size(self):
         # 5MB is minimum part size
         part_size = 8388608
         file_size = 5000
-        self._run_multipart_test(part_size, file_size)
+        return self._run_multipart_test(part_size, file_size)
```
Comment (Tarrasch) [inline test/contrib/s3_test.py]: Why do you return the last thing you do in a test case??? I thought you only used self.assertBlah()

Comment (dlstadther) [reply test/contrib/s3_test.py]: The `test_put_multipart...` tests are used below as args to `self._run_copy_test(...)` and I need the output of `self._run_multipart_test` to verify if the correct num and size of files were copied
